### PR TITLE
BrowserStack Karma configuration

### PR DESCRIPTION
**Pull request by danielnaab**
Before making a pull request
----------------------------

- [x] Make sure all tests run
- [x] Update the changelog appropriately

WDFN-278
-----------
This PR adds some BrowserStack browsers to the Karma configuration, and fixes a small number of bugs found during the process of testing it.

Additionally, if the credentials are available, TravisCI was configured to run against the BrowserStack browsers. Unfortunately, pull requests from forks do not have access to encrypted TravisCI variables, for security reasons. As a result, this PR will fallback to just local Firefox in that circumstance. We'll need to merge to master to confirm that that complete suite is run successfully when merged.

After making a pull request
---------------------------
- [x] If appropriate, put the link to the PR in the JIRA ticket
- [x] Assign someone to review unless the change is trivial

### CODE REVIEW

File: assets/karma.conf.js [hunk 1]
```diff
@@ -75,20 +75,91 @@ module.exports = function (config) {
...
         singleRun: true,
 
         // Concurrency level
         // how many browser should be started simultaneous
         concurrency: Infinity
-    });
+    };
+
+    // Add BrowserStack-specific settings
+    if (process.env.KARMA_BROWSERSTACK && process.env.BROWSER_STACK_USERNAME) {
```
Comment (mbucknell) [inline assets/karma.conf.js]: How does BROWSER_STACK_USERNAME get set?

Comment (danielnaab) [reply assets/karma.conf.js]: Via the secret environment variables in .travisci (or set locally)

Comment (mbucknell) [reply assets/karma.conf.js]: I don't understand ... do you have some links that could explain what's going on?

Comment (danielnaab) [reply assets/karma.conf.js]: Sure, here are the docs on that: https://docs.travis-ci.com/user/environment-variables/#Defining-encrypted-variables-in-.travis.yml

And a discussion of the problems of using it via forks: https://github.com/travis-ci/travis-ci/issues/1946

Comment (mbucknell) [reply assets/karma.conf.js]: I think I found it. It's in the .travis file. I got confused by the .travisci

Comment (danielnaab) [reply assets/karma.conf.js]: Oops, sorry 😄

**Pull request merged**